tests/virt_autotest/xen_guest_irqbalance.pm

```diff
@@ -0,0 +1,159 @@
+# SUSE's openQA tests
+
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# Summary: xen domU irqbalance test
+# Maintainer: Julie CAO <JCao@suse.com>
+
+use base "virt_feature_test_base";
+use strict;
+use warnings;
+use testapi;
+use version_utils qw(is_sle);
+use set_config_as_glue;
+use virt_autotest::common;
+use virt_autotest::utils qw(is_kvm_host guest_is_sle wait_guest_online print_cmd_output_to_file);
+
+sub run_test {
+    my $self = shift;
+
+    return if is_kvm_host;
+    #save original guest configuration file in case of restore in post_fail_hook()
+    my $downloaded_xml_dir = "/tmp/download_vm_xml";
+    foreach my $guest (keys %virt_autotest::common::guests) {
+        unless (script_run("ls $downloaded_xml_dir/$guest.xml") == 0) {
+            assert_script_run "mkdir -p $downloaded_xml_dir";
+            assert_script_run "virsh dumpxml --inactive $guest > $downloaded_xml_dir/$guest.xml";
+        }
+    }
+
+    foreach my $guest (keys %virt_autotest::common::guests) {
+
+        #irqbalance test run on sle12sp3+ guest
+        return if guest_is_sle($guest, '<=12-sp2');
+
+        record_info("Test $guest");
+        prepare_guest_for_irqbalance($guest);
+
+        my $nproc = script_output("ssh root\@$guest 'nproc'");
+
+        #Check if network interrupts are distributed over CPUs in guest initally as irqbalance is enabled by default
```

Comment:
Can we add a check about  irqbalance service status? This can help if the default service state changes or by some bug the service can not be enabled automatically.

Reply:
we have had that check in sub prepare_guest_for_irqbalance().